### PR TITLE
[4.0] Changing npm to use browserslist directive

### PR DESCRIPTION
**Pull request by Hackwar**
A few days ago the npm step of our Drone CI displayed messages that we should use browserslist to configure our browsers dependencies. This does that.

### How to test?
Apply patch, run npm i and test if all scripts are still created correctly.

### CODE REVIEW

**Comment by wilsonge** [comment]
@Hackwar you need to fix the js lint errors

**Comment by ghazal** [comment]
I have tested this item :white_check_mark: successfully on 66fd3f31e000fb06699bbdd9e2056bbcb544578a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25557">issues.joomla.org/tracker/joomla-cms/25557</a>.</sub>

File: build/build-modules-js/javascript/compile-w-c.es6.js [hunk 1]
```diff
@@ -77,9 +77,8 @@ const createJsFiles = (inputFile, es6FileContents) => {
  * Compiles any web component/custom element files from the media_source folder
  *
  * @param file     The full path to the file + filename + extension
- * @param options  The options from the settings.json
  */
-module.exports.compile = (inputFile, options) => {
```
Comment (wilsonge) [inline build/build-modules-js/javascript/compile-w-c.es6.js:82]: You need to find where this is called (`handle-file.es6.js`) and not pass options in there too please ;)

File: build/build-modules-js/stylesheets/scss-transform.es6.js [hunk 2]
```diff
@@ -6,7 +6,7 @@ const Postcss = require('postcss');
 const Sass = require('node-sass');
 const MakeDir = require('../utils/make-dir.es6.js');
 
-module.exports.compile = (file, options) => {
```
Comment (wilsonge) [inline build/build-modules-js/stylesheets/scss-transform.es6.js:9]: Same here

**Comment by Hackwar** [comment]
Fixed your remarks

**Comment by Hackwar** [comment]
So much for "fixed that". Still more to fix...

**Comment by Hackwar** [comment]
Seems like I found everything in the last commit. RTC?

**Comment by wilsonge** [comment]
Thanks :)

**Pull request merged**